server/src/main/java/io/crate/types/StringType.java

```diff
@@ -142,6 +153,26 @@ public String explicitCast(Object value) throws IllegalArgumentException, ClassC
         }
     }
 
+    @Override
+    public String valueForInsert(Object value) {
+        if (value == null) {
+            return null;
+        }
+        assert value instanceof String
+            : "valueForInsert must be called only on objects of String type";
+        var string = (String) value;
+        if (unbound() || string.length() <= lengthLimit) {
+            return string;
+        } else {
+            if (isBlank(string, lengthLimit, string.length())) {
+                return string.substring(0, lengthLimit);
+            } else {
+                throw new IllegalArgumentException(
+                    "The value is too long for text type of the length: " + lengthLimit);
```

Comment:
Maybe we could even include the value in the error message

Reply:
The whole value or maybe trim it after few characters, e.g. `fasfdsf... is to long for ...`

Reply:
Yes, trimming after a certain length may make sense. After 20 chars or so?

Reply:
20 sounds good 👍